File: user-sites/user_2035/bf.py
#!/usr/bin/env python3
import sys

def run(code):
    tape = [0] * 30000
    p = 0
    ip = 0
    while ip < len(code):
        c = code[ip]
        if c == '>': p += 1
        elif c == '<': p -= 1
        elif c == '+': tape[p] = (tape[p] + 1) & 255
        elif c == '-': tape[p] = (tape[p] - 1) & 255
        elif c == '.': sys.stdout.write(chr(tape[p]))
        elif c == ',': tape[p] = ord(sys.stdin.read(1)) if not sys.stdin.isatty() else 0
        elif c == '[':
            if tape[p] == 0:
                depth = 1
                while depth:
                    ip += 1
                    if code[ip] == '[': depth += 1
                    if code[ip] == ']': depth -= 1
        elif c == ']':
            if tape[p] != 0:
                depth = 1
                while depth:
                    ip -= 1
                    if code[ip] == ']': depth += 1
                    if code[ip] == '[': depth -= 1
        ip += 1

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("usage: bf.py <code>", file=sys.stderr)
        sys.exit(1)
    run(sys.argv[1])

File: user-sites/user_2035/bf.c
#include <stdio.h>
#include <stdlib.h>

char tape[30000];
int p = 0;

void run(char *code) {
    char *ip = code;
    while (*ip) {
        switch (*ip) {
            case '>': p++; break;
            case '<': p--; break;
            case '+': tape[p]++; break;
            case '-': tape[p]--; break;
            case '.': putchar(tape[p]); break;
            case ',': tape[p] = getchar(); break;
            case '[': {
                if (!tape[p]) {
                    int depth = 1;
                    while (depth) {
                        ip++;
                        if (*ip == '[') depth++;
                        if (*ip == ']') depth--;
                    }
                }
                break;
            }
            case ']': {
                if (tape[p]) {
                    int depth = 1;
                    while (depth) {
                        ip--;
                        if (*ip == ']') depth++;
                        if (*ip == '[') depth--;
                    }
                }
                break;
            }
        }
        ip++;
    }
}

int main(int argc, char **argv) {
    if (argc < 2) {
        fprintf(stderr, "usage: bf <code>\n");
        return 1;
    }
    run(argv[1]);
    return 0;
}
